lib/facility.rb

```diff
@@ -1,14 +1,59 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :collected_fees, :registered_vehicles, :registration_date
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_hash)
+    @name = facility_hash[:name]
+    @address = facility_hash[:address]
+    @phone = facility_hash[:phone]
     @services = []
+    @collected_fees = 0
+    @registered_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration") 
+      @registered_vehicles << vehicle 
+      vehicle.registration_date = Date.today
+      collect_fees(vehicle)
+      @registered_vehicles
+    end
```

Comment:
Nice job here including all the logic for all the side effects this method needed to accomplish. I like that you abstracted some of that logic into helper methods to keep this method very clean and readable.

Reply:
Actually, we're missing one thing! This method should also be setting the plate_type of our vehicle.